recipes/sdl/all/conanfile.py

```diff
@@ -321,17 +273,58 @@ def define_toolchain(self):
         elif self.settings.os == "Windows":
             tc.variables["SDL_DIRECTX"] = self.options.directx
 
-        if Version(self.version) >= "2.0.22":
-            tc.variables["SDL2_DISABLE_SDL2MAIN"] = not self.options.sdl2main
+        tc.variables["SDL2_DISABLE_SDL2MAIN"] = not self.options.sdl2main
 
         # Add extra information collected from the deps
-        tc.variables["EXTRA_LDFLAGS"] = ";".join(cmake_extra_ldflags)
-        tc.variables["CMAKE_REQUIRED_INCLUDES"] = ";".join(cmake_required_includes)
-        cmake_extra_cflags = ["-I{}".format(path) for _, dep in self.dependencies.items() for path in dep.cpp_info.includedirs]
-        tc.variables["EXTRA_CFLAGS"] = ";".join(cmake_extra_cflags).replace(os.sep, '/')
-        tc.variables["EXTRA_LIBS"] = ";".join(cmake_extra_libs)
+        all_deps = [dep for dep in reversed(self.dependencies.host.topological_sort.values())]
+        if not self._is_cl_frontend:
+            deps_includes = [p.replace("\\", "/") for dep in all_deps for p in dep.cpp_info.aggregated_components().includedirs]
+            tc.variables["CMAKE_REQUIRED_INCLUDES"] = ";".join(deps_includes)
+            extra_cflags = [f"-I{p}" for p in deps_includes]
+            extra_cflags += [f"-D{define}" for dep in all_deps for define in dep.cpp_info.aggregated_components().defines]
+            tc.variables["EXTRA_CFLAGS"] = ";".join(extra_cflags)
+            extra_ldflags = ["-L{}".format(p.replace("\\", "/")) for dep in all_deps for p in dep.cpp_info.aggregated_components().libdirs]
+            tc.variables["EXTRA_LDFLAGS"] = ";".join(extra_ldflags)
+            extra_libs = [lib for dep in all_deps for lib in dep.cpp_info.aggregated_components().libs]
+            extra_libs += [lib for dep in all_deps for lib in dep.cpp_info.aggregated_components().system_libs]
+            tc.variables["EXTRA_LIBS"] = ";".join(extra_libs)
+
         tc.generate()
 
+        CMakeDeps(self).generate()
+        PkgConfigDeps(self).generate()
+
+        env = Environment()
+        lib_paths = [lib for dep in all_deps for lib in dep.cpp_info.aggregated_components().libdirs]
+        env.define_path("LIBRARY_PATH", os.pathsep.join(lib_paths))
+        env.vars(self).save_script("sdl_library_path")
+
+    def _patch_sources(self):
+        cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
+        if self.settings.os == "Macos":
```

Comment:
```suggestion
        if is_apple_os(self):
```
Can be generalized to other Apple OS-s, perhaps.